### PR TITLE
Adds throat slicing

### PR DESCRIPTION
**Pull request by Fhaxaris**
## About The Pull Request
Lets you cut someone's throat under certain conditions. Higher damage weapons get the job done faster. Weapon must be sharp, of course.
- Neck grab
- Critical
- Unconsciousness

The bleeding rate is 2.5x Heparin, for comparison.

For me at least, it is a little annoying that there is no way to execute someone in a quick, decisive action with a melee weapon and I am instead stuck stabbing them until they go horizontal. I can go get a gun and do the special execution action, but that's no fun.

## Changelog
:cl:
add: You cut slice peoples throats now. Select the head on harm intent while they are sleeping, in critical, or neck grabbed.
/:cl:

### CODE REVIEW

**Comment by spessbro** [comment]
antag freeze :(

**Comment by Fhaxaris** [comment]
what does this have anything to do with the antag freeze

**Comment by nicbn** [comment]
Nice

**Comment by spessbro** [comment]
lethal content is mostly meant for antags i don't think this will pass but cool

**Comment by Tharcoonvagh** [comment]
assistants exist

**Comment by Tlaltecuhtli** [comment]
time to pen throat slicing and clone

**Comment by actioninja** [comment]
Eh, while lethal mechanics are generally used more by antags they're not really _exclusively_ antag mechanics.
On top of that, the spirit of the antag freeze is more about obsessive and constant balancing of antag-only items, constant gamemode tweaking, and adding yet more stupid meme antag items.

**Comment by necromanceranne** [comment]
CQC buff. Immense CQC buff. Sleeping everyone then slitting their throats would be easy peezy with an esword or just even the knife nukies start with.

Also the chef actually. A sharpened cleaver will prove to be a fairly scary weapon.

**Comment by carlarctg** [comment]
CQC is nukie only, by the time a nukie slits your throat they could very easily have used literally any of their other weapons to kill you too
chefs can still only use CQC in the kitchen so i don't know what youre on about

**Comment by necromanceranne** [comment]
You enter the kitchen. I sleep you and slit your throat. There is no saving you. You died in a single decisive blow.

Also, while there is some truth to that the problem is that this is a fairly quiet method too. Sleep is one of the more quiet moves CQC has where as conseq. CQC and headkicks tend to be rather loud. Depending on how loud the throat slice is (I doubt it even makes a noise), that makes for an easy stealth kill or two.

Not necessarily a complaint. Just something to note.

**Comment by kriskog** [comment]
This doesn't instantly kill you, it does weapon brute damage and makes you lose blood. Blood loss is kinda underwhelming compared to just beating the shit out of someone some more. If people are already in crit and taking oxyloss from that, this won't make a big difference other than make them slightly harder to save afterwards. But not really, since stasis beds.

**Comment by nemvar** [comment]
I don't see the point in this. It just bloats the already kinda shitty butchering component and what is the gameplay gain? In the vast majority of cases, this is probably slower than just bashing the guys head in.

**Comment by Tlaltecuhtli** [comment]
you can rp as terrorist and do executions live

File: code/datums/components/butchering.dm [hunk 1]
```diff
@@ -27,13 +27,29 @@
...
+/datum/component/butchering/proc/startNeckSlice(obj/item/source, mob/living/carbon/human/H, mob/living/user)
+	H.visible_message("<span class='danger'>[user] is slitting [H]'s throat!</span>", \
+					"<span class='userdanger'>Your throat is being slit by [user]!</span>", \
+					"<span class='userdanger'>Something is cutting into your neck!</span>")
+	playsound(H.loc, butcher_sound, 50, TRUE, -1)
+	if(do_mob(user, H, CLAMP(500 / source.force, 30, 50)) && H.Adjacent(source))
+		H.visible_message("<span class='danger'>[user] slits [H]'s throat!</span>", \
+					"<span class='userdanger'>[user] slits your throat...</span>")
+		H.adjustBruteLoss(source.force, 0)
+		H.bleed_rate = CLAMP(H.bleed_rate + 20, 0, 30)
```
Comment (KetasDotepasto) [inline code/datums/components/butchering.dm]: A bleed rate of 20 will result in the victim loosing 390 units of blood, not accounting for natural blood regeneration, which, over 40 ticks, ammounts to 20 units of blood, assuming the victim is well fed. So we are down to 370 units of blood lost. Normal blood volume is 560, meaning the victim will have 190 units of blood left once he stops bleeding.

190 units of blood falls into BLOOD_VOLUME_BAD, which causes 5 units of oxyloss per tick. Respiration restores 5 units every 4 ticks. In addition the victim has a 15% chance per tick of falling unconscious, will therefore happen once every 7 ticks.

At 3.75 oxyloss per tick (this accounts natural oxyloss regeneration), it takes 54 ticks to die. This neglects the fact that the victim will be dying faster once he hits crit. It also neglets the brute damage done by this maneuver. During 54 ticks, the victim will generate 27 units of blood, bringing him up 217 units, which is still BLOOD_VOLUME_BAD.

We will hit BLOOD_VOLUME_BAD 25 ticks in. Neglecting the oxyloss that occurs before BLOOD_VOLUME_BAD, getting your throat slit will kill you in 79 ticks. Asuming 1.3 seconds per tick (The brain tumor quirk needs 1000 ticks to kill you, which I know from experience to be about 22 minutes), this will kill you in less than 1 minute and 40 seconds.

Of course, the bandages, an item available in the white medkit, will stop all bleeding. So, assuming the victim escapes immediately after his or her throat has been slit, they have about 30 seconds to find bandages. Best equipped to help the victim would be a mediborg, who has bandages, to stop the bleeding, saline-glucose solution, to reduce the severity of bloodloss, lollipops with iron and omnizine to both replace blood, and cure oxyloss damage.

Getting your throat slit is guranteed death, unless you have bandages on your person, or receive immediate medical aid. Another approach would be taking heparin, which caps your bleed rate at 8, which results in only 60 units of blood lost, assuming it is in your system for 1 tick only.

**Review by KetasDotepasto — APPROVED** [review]
I have analyzed your choice of bleed rate. Overall, I can confidently state that this will kill anybody who gets their throat slit quickly, but still allows an EMT or mediborg to save the victim. More importantly, unless the victim has prepared for having their throat slit beforehand, they'll be certainly unable to save themselves.

**Comment by CRITAWAKETS** [comment]
[IN THE NAME OF KANE!](https://www.youtube.com/watch?v=DZrGIIIBhNg)

Shitpost aside, this seems like a pretty useful feature for executing prisoners.

**Comment by Fhaxaris** [comment]
> Now technically, same with CQC in of itself but people can hear and catch you beating the shit out of someone with CQC.

You can sleep someone with CQC, and set them on fire. They are burning to death and can't do anything about it. Burning is also completely silent, except for the initial ignition. They also can't be cloned without restoring the burn damage. You can sleep someone with CQC, and overdose them on meds. Pretty simple and silent. You can sleep someone with CQC, and buckle cuff them in the freezer, They freeze to death, and it's silent if you take their radio. You can sleep someone with CQC, and give them an internals tank with whatever you want and open the valve. (please correct me if any of these are incorrect, I haven't actually tested them)

There are probably hundreds of ways chefs can kill people with their CQC ability. The core problem is that CQC is a martial art with ridiculous powergaming potential. If the chef wants to go on a mass murdering spree with CQC, nothing is stopping them from sleeping you and doing whatever they want with you. No one is going to combine CQC and neck slicing in a way that is any different then the methods I just listed.

tl;dr: there are several ways to silently murder tons of people when using CQC; powergamers are going to powergame, this doesn't actually make it easier.

**Comment by Fhaxaris** [comment]
> this is probably slower than just bashing the guys head in.

bashing the guys head in has no flavour

**Comment by KetasDotepasto** [comment]
Fountain pens are sharp.

This would also allow for the following scenario: You are the captain. Shift has just started, you pick up your spare, and the nuke disk. A greytider breaks into your office. You subdue him with your energy gun, grap him by the neck. Then you take out your fountain pen, and slit his throat. You throw him against the wall, before locking him in your office. You make a mental note to have the janitor clean up his body as you go for your morning coffee.

Just think of the roleplay opportunities this PR would provide.

**Comment by Fhaxaris** [comment]
Note to self: change Max on clamp from 50 to 100

**Comment by MMMiracles** [comment]
Guns allow for special-case executions so I don't see the issue with there being the melee equivalent as an option.

**Comment by PKPenguin321** [comment]
As long as this takes as long or longer than gun executions, sure. Sharp weapons are a lot more common than guns, you see.

**Comment by Geikkamir** [comment]
> You enter the kitchen. I sleep you and slit your throat. There is no saving you. You died in a single decisive blow. Now technically, same with CQC in of itself but people can hear and catch you beating the shit out of someone with CQC.
> 
> Also, while there is some truth to nukies just gunning you down with all kinds of weapons, the problem is that this is a fairly quiet method too. Sleep is one of the more quiet moves CQC has where as conseq. CQC and headkicks tend to be rather loud. Depending on how loud the throat slice is (I doubt it even makes a noise), that makes for an easy stealth kill or two.
> 
> Not necessarily a complaint. Just something to note.

May I remind you that the kitchen literally has a gibber in the back room.

I don't think throat slicing is going to be more effective that KO>stuff in gibber

**Comment by carlarctg** [comment]
throatslicing should at the very MOST be equal or slightly higher to the amount of time it takes to kill someone (deal 200 brute) with your weapon

if you take 10 seconds to throat cut with a 15 damage survival knife, you could just as easily have killed them in around 7 seconds total instead, ruining the point of the throat-cut altogether as people will rarely use it

not sure doubling low force cut time is a good idea because of that, 100 is double the type it takes to fireman carry right? that's like, a LOT of time to spend cutting someone's neck when you could have beheaded them with a sharp weapon and pulled them all the way to a maint crate or something, unless i'm reading code wrong again

**Comment by KetasDotepasto** [comment]
@PKPenguin321 It takes quite a while after the throat is slit for the victim to actually die, so I don't think this, as it is now, is overpowered compared to the gunshot execution, considering that if you are asleep, or in crit, or grabbed by the neck, you're already pretty fucked if somebody wants to kill you even without throat slicing.

**Comment by Fhaxaris** [comment]
> 100 is double the type it takes to fireman carry right? that's like, a LOT of time to spend cutting someone's neck

this only applies for weapons with a force <10, which I don't think can even dismember


For reference: Bruise packs take 20, fireman carry takes 50.
Captain's pen, glass shard, 5 force: 500/5 == 100
Kitchen knife, bone knife, 10 force: 500/10 == 50
Butcher's knife, survival knife, 15 force: 500/15 == 33
Edagger, 18 force: 500/18 == 27 (30)
Combat knife, 20 force: 500/20 == 25 (30)

**Comment by Geikkamir** [comment]
> throatslicing should at the very MOST be equal or slightly higher to the amount of time it takes to kill someone (deal 200 brute) with your weapon
> 
> if you take 10 seconds to throat cut with a 15 damage survival knife, you could just as easily have killed them in around 7 seconds total instead, ruining the point of the throat-cut altogether as people will rarely use it
> 
> not sure doubling low force cut time is a good idea because of that, 100 is double the type it takes to fireman carry right? that's like, a LOT of time to spend cutting someone's neck when you could have beheaded them with a sharp weapon and pulled them all the way to a maint crate or something, unless i'm reading code wrong again

Keep in mind that this is being pitched as essentially the melee equivalent to the execution move with guns. In that sense, it's kind of fine for it to be less viable than just bashing them to death with the melee weapon. Keep in mind that 99% of the time you could literally beat the victim to death with your gun faster than you could kill them with the execution. It's fine for it to be something you do more for style than power.

File: code/datums/components/butchering.dm [hunk 2]
```diff
@@ -27,13 +27,29 @@
...
+/datum/component/butchering/proc/startNeckSlice(obj/item/source, mob/living/carbon/human/H, mob/living/user)
+	H.visible_message("<span class='danger'>[user] is slitting [H]'s throat!</span>", \
+					"<span class='userdanger'>Your throat is being slit by [user]!</span>", \
+					"<span class='userdanger'>Something is cutting into your neck!</span>")
+	playsound(H.loc, butcher_sound, 50, TRUE, -1)
+	if(do_mob(user, H, CLAMP(500 / source.force, 30, 100)) && H.Adjacent(source))
+		H.visible_message("<span class='danger'>[user] slits [H]'s throat!</span>", \
+					"<span class='userdanger'>[user] slits your throat...</span>")
+		H.adjustBruteLoss(source.force, 0)
+		H.bleed_rate = CLAMP(H.bleed_rate + 20, 0, 30)
```
Comment (nemvar) [inline code/datums/components/butchering.dm]: If you want flavour, make the guy gurgle/gasp and/or some other emotes after getting their sloat slit, preferably on a timer.

File: code/datums/components/butchering.dm [hunk 3]
```diff
@@ -27,13 +27,29 @@
...
 /datum/component/butchering/proc/startButcher(obj/item/source, mob/living/M, mob/living/user)
 	to_chat(user, "<span class='notice'>You begin to butcher [M]...</span>")
 	playsound(M.loc, butcher_sound, 50, TRUE, -1)
 	if(do_mob(user, M, speed) && M.Adjacent(source))
 		Butcher(user, M)
 
+/datum/component/butchering/proc/startNeckSlice(obj/item/source, mob/living/carbon/human/H, mob/living/user)
+	H.visible_message("<span class='danger'>[user] is slitting [H]'s throat!</span>", \
+					"<span class='userdanger'>Your throat is being slit by [user]!</span>", \
+					"<span class='userdanger'>Something is cutting into your neck!</span>")
```
Comment (nemvar) [inline code/datums/components/butchering.dm]: blind_message is shown to all mobs nearby so this doesn't really make sense.
```
	user.visible_message("<span class='danger'>[user] is slitting [H]'s throat!</span>", \
					"<span class='danger'>You start slicing [H]'s throat!</span>", \
					"<span class='hear'>You hear a cutting noise!</span>", ignored_mobs = H)
	H.show_message("<span class='userdanger'>Your throat is being slit by [user]!</span>", MSG_VISUAL, "<span class = 'userdanger'>Something is cutting into your neck!</span>", NONE)
```
This properly covers ALL cases (which tbh is better than most messages)

File: code/datums/components/butchering.dm [hunk 4]
```diff
@@ -27,13 +27,29 @@
...
+	playsound(H.loc, butcher_sound, 50, TRUE, -1)
+	if(do_mob(user, H, CLAMP(500 / source.force, 30, 100)) && H.Adjacent(source))
+		H.visible_message("<span class='danger'>[user] slits [H]'s throat!</span>", \
+					"<span class='userdanger'>[user] slits your throat...</span>")
+		H.adjustBruteLoss(source.force, 0)
```
Comment (nemvar) [inline code/datums/components/butchering.dm]: Fairly certain the 0 should be a FALSE and it also shouldn't be there at all because we WANT health to update here.
Should also be apply_damage() instead of adjustBruteLoss because this is targeted to the head.

File: code/datums/components/butchering.dm (hunk 3 above)
Comment (Fhaxaris) [reply code/datums/components/butchering.dm]: Oops, I miss read the visible_message code. Thank you for this.

File: code/datums/components/butchering.dm (hunk 2 above)
Comment (Fhaxaris) [reply code/datums/components/butchering.dm]: That is a good idea. I am new to DM, my best idea is to make a new status effect in the status_effect datum.

**Comment by Learner10** [comment]
If I'm a chef and I throat slice someone who's standing on top of a bucket, do I get to keep the blood to use in stews or drinks?

**Comment by Fhaxaris** [comment]
>If I'm a chef and I throat slice someone who's standing on top of a bucket, do I get to keep the blood to use in stews or drinks?


I don't know how blood loss works. If you are bleeding and above a bucket, does your blood go in? If so, then yes you could.

**Comment by kriskog** [comment]
> 
> 
> If I'm a chef and I throat slice someone who's standing on top of a bucket, do I get to keep the blood to use in stews or drinks?

Bit out of scope, but someone could add blood "milking" when slicing the throat of someone on a meathook.

**Comment by actioninja** [comment]
Yeah, I like weird little interactions like this. This most likely won't even be used that much.
